Source/DafnyLanguageServer/Workspace/ICompilationStatusNotificationPublisher.cs

```diff
@@ -12,6 +12,7 @@ public interface ICompilationStatusNotificationPublisher {
     /// </summary>
     /// <param name="textDocument">The document which had a compilation status change.</param>
     /// <param name="status"></param>
-    void SendStatusNotification(TextDocumentItem textDocument, CompilationStatus status);
+    /// <param name="message">Additional info about the current status</param>
+    void SendStatusNotification(TextDocumentItem textDocument, CompilationStatus status, string? message = null);
```

Comment:
Note that we'll run into some versioning issues when we change this API again.

The Dafny VSCode extension should be compatible with old and new Dafny versions, so it'll have to handle both the version of this API that this PR is adding, and the future API which will be different.

Since we don't have many customer yet, we might be able to make an exception though, and say that from a certain version of the extension it only supports Dafny 3.5 (or whatever version we add the final API in) or higher. Maybe that's the most practical approach here.

Reply:
That's right, and that's also why I stuck with very very basic progress report here: since the cost on the IDE side to support this is literally below 10 lines, keeping support for it while supporting a more advanced API in the future will be a very small cost.